content/guides/quickstart/voting-guide.md

```diff
@@ -0,0 +1,1230 @@
+        %metadata-store
+        (scot %da now.bol)
+        /metadata/groups/ship/(scot %p p.gid)/[q.gid]/noun
+      ==
+    ?~  group-data
+      [gid q.gid]
+    [gid title.metadatum.u.group-data]
+  |=([[* a=@] [* b=@]] (aor a b))
+::
+++  all-group-names
+  ^-  (list (pair gid @t))
+  (get-group-names ~(tap in get-groups))
+::
+++  get-groups
+  ^-  (set gid)
+  %.  head
+  %~  run  in
+  %.  %groups
+  %~  get  ju
+  .^  (jug app-name:ms [gid *])
+    %gy
+    (scot %p our.bol)
+    %metadata-store
+    (scot %da now.bol)
+    /app-indices
+  ==
+++  our-life
+  .^  life
+    %j
+    (scot %p our.bol)
+    %life
+    (scot %da now.bol)
+    /(scot %p our.bol)
+  ==
+--
+```
+
+### Desk config
+
+With our types, agent, mark files and front-end now complete, the last thing we
+need are some desk configuration files.
+
+Firstly, we need to specify the kernel version our app is compatible with. We do
+this by adding a `sys.kelvin` file to the root of our `tally` directory:
+
+```shell
+cd tally
+echo "[%zuse 418]" > sys.kelvin
+```
+
+We also need to specify which agents to start when our desk is installed. We do
+this in a `desk.bill` file:
+
+```shell
+echo "~[%tally]" > desk.bill
+```
+
+Lastly, we need to create a Docket file. Docket is the agent that manages app
+front-ends - it fetches & serves them, and it also configures the app tile and
+other metadata. Create a `desk.docket-0` file in the `tally` directory and add the
+following:
+
+```shell
+:~
+  title+'Tally'
+  info+'Ring signature voting for groups.'
+  color+0xc4.251a
+  version+[0 1 0]
+  website+'https://urbit.org'
+  license+'MIT'
+  base+'tally'
+  site+/tally
+==
+```
+
+### Put it together
+
+Our app is now complete, so let's try it out. In the Dojo of our fake ~zod,
+we'll create a new desk by forking from an existing one:
+
+```
+|merge %tally our %webterm
+```
+
+Next, we'll mount the desk so we can access it from the host OS:
+
+```
+|mount %tally
+```
+
+Currently its contents are the same as the `%webterm` desk, so we'll need to
+delete those files and copy in our own instead. In the normal shell, do the
+following:
+
+```bash
+rm -r zod/tally/*
+cp -r tally/* zod/tally/*
+```
+
+Back in the Dojo again, we can now commit those files and install the app:
+
+```
+|commit %tally
+|install our %tally
+```
+
+If we open a web browser and go to `localhost:8080`, we should see a tile for
+the Tally app. If we click on it, it'll open our React front-end and we can start
+using it.
+
+Once we've confirmed it's working on our fake ~zod, we can try installing it on
+a real ship. To do that, we can just repeat the steps in this section with our
+actual ship. On the live network, we can also publish the app so others can
+install it from us. To do so, just run the following command:
+
+```
+:treaty|publish %tally
+```
+
+Now our friends will be able to install it with `|install <our ship> %tally`
+or by searching for `<our ship>` on their ship's homescreen.
```

Comment:
“Ships are intended to run continuously, but fake ships for development are frequently stopped between sessions.  Use `|exit` to shut down your fake ship.”